packages/cli/src/zosjobs/submit/Submit.shared.handler.ts

```diff
@@ -31,150 +31,164 @@ export default class SharedSubmitHandler extends ZosmfBaseHandler {
+                        const getJcl = await Get.dataSet(this.mSession, this.mArguments.dataset, options);
+                        source = this.mArguments.dataset;
+
+                        apiObj = await SubmitJobs.submitJclString(this.mSession, getJcl.toString(), parms);
+                        if (parms.viewAllSpoolContent) {
+                            spoolFilesResponse = apiObj;
+                        }
+
+                        break;
+                    } else {
+                        response = await SubmitJobs.submitJobCommon(this.mSession, {jobDataSet: this.mArguments.dataset,
+                            jclSymbols: this.mArguments.jclSymbols});
+                        apiObj = await SubmitJobs.checkSubmitOptions(this.mSession, parms, response);
+                        source = this.mArguments.dataset;
+
+                        if (parms.viewAllSpoolContent) {
+                            spoolFilesResponse = apiObj;
+                        }
                     }
 
                     break;
-                } else {
-                    response = await SubmitJobs.submitJobCommon(this.mSession, {jobDataSet: this.mArguments.dataset,
+                // Submit JCL from a USS file
+                case "uss-file":
+                    response = await SubmitJobs.submitJobCommon(this.mSession, {jobUSSFile: this.mArguments.file,
                         jclSymbols: this.mArguments.jclSymbols});
                     apiObj = await SubmitJobs.checkSubmitOptions(this.mSession, parms, response);
-                    source = this.mArguments.dataset;
+                    source = this.mArguments.ussfile;
 
                     if (parms.viewAllSpoolContent) {
                         spoolFilesResponse = apiObj;
                     }
-                }
 
-                break;
-            // Submit JCL from a USS file
-            case "uss-file":
-                response = await SubmitJobs.submitJobCommon(this.mSession, {jobUSSFile: this.mArguments.file,
-                    jclSymbols: this.mArguments.jclSymbols});
-                apiObj = await SubmitJobs.checkSubmitOptions(this.mSession, parms, response);
-                source = this.mArguments.ussfile;
-
-                if (parms.viewAllSpoolContent) {
-                    spoolFilesResponse = apiObj;
-                }
+                    break;
+                // Submit the JCL from a local file
+                case "local-file": {
+                    parms.jclSource = this.mArguments.localFile;
+                    try {
+                        const JclString = fs.readFileSync(this.mArguments.localFile).toString();
+                        apiObj = await SubmitJobs.submitJclString(this.mSession, JclString, parms);
+                        source = this.mArguments.localFile;
+                    } catch (err) {
+                        throw new ImperativeError({
+                            msg: ZosFilesMessages.nodeJsFsError.message,
```

Comment:
Please copy the definition for `nodeJsFsError` and add it to ZosJobsMessages so it can be imported from there instead.